apiserver/facades/agent/proxyupdater/proxyupdater.go

```diff
@@ -16,6 +17,71 @@ import (
 	"github.com/juju/juju/state/watcher"
 )
 
+// ProxyUpdaterV1 defines the pubic methods for the v1 facade.
+type ProxyUpdaterV1 interface {
+	ProxyConfig(args params.Entities) params.ProxyConfigResultsV1
+	WatchForProxyConfigAndAPIHostPortChanges(args params.Entities) params.NotifyWatchResults
+}
+
+var _ ProxyUpdaterV1 = (*APIv1)(nil)
+
+// ProxyUpdaterV2 defines the pubic methods for the v2 facade.
+type ProxyUpdaterV2 interface {
+	ProxyConfig(args params.Entities) params.ProxyConfigResults
+	WatchForProxyConfigAndAPIHostPortChanges(args params.Entities) params.NotifyWatchResults
+}
+
+var _ ProxyUpdaterV2 = (*APIv2)(nil)
+
+// NewFacadeV1 provides the signature required for facade registration
+// and creates a v1 facade.
+func NewFacadeV1(ctx facade.Context) (*APIv1, error) {
+	api, err := NewFacadeV2(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &APIv1{api}, nil
+}
+
+// NewFacadeV2 provides the signature required for facade registration
+// and creates a v2 facade.
+func NewFacadeV2(ctx facade.Context) (*APIv2, error) {
+	api, err := newFacadeBase(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &APIv2{api}, nil
+}
+
+func newFacadeBase(ctx facade.Context) (*APIBase, error) {
+	st := ctx.State()
+	m, err := st.Model()
+	if err != nil {
+		return nil, err
+	}
+	return NewAPIBase(
+		&stateShim{st: st, m: m},
+		ctx.Resources(),
+		ctx.Auth(),
+	)
+}
+
+// APIv1 provides the ProxyUpdater version 1 facade.
+type APIv1 struct {
+	*APIv2
+}
+
+// APIv2 provides the ProxyUpdater version 2 facade.
+type APIv2 struct {
+	*APIBase
+}
+
+type APIBase struct {
```

Comment:
comment
or, does APIBase really need to be exported

Reply:
Not strictly, but it does mean we can use them without export_test wrappers.